### PR TITLE
dockerfiles: upgrade to ubuntu:20.04; some chore

### PR DESCRIPTION
**Pull request by TriplEight**
I'm upgrading the CI image from being based on `debian:buster` to `ubuntu:20.04`, some details: https://github.com/paritytech/scripts/pull/309
The binaries produced in our new CI image will be linked by the newer `GLIBC` so won't be executable in debian:
```
/usr/local/bin/adder-collator: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /usr/local/bin/adder-collator)
/usr/local/bin/adder-collator: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /usr/local/bin/adder-collator)
```

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: bridges/deployments/networks/OpenEthereum.Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:xenial AS builder
```
Comment (TriplEight) [inline bridges/deployments/networks/OpenEthereum.Dockerfile:1]: @tomusdrw is this image relevant to Polkadot?
Also, Xenial (16.04) is out of standard support since April 2021, our containers policy is not happy.

File: scripts/docker/release.Dockerfile [hunk 2]
```diff
@@ -19,24 +19,25 @@ ENV RUST_BACKTRACE 1
...
-		useradd -m -u 1000 -U -s /bin/sh -d /polkadot polkadot && \
-		gpg --recv-keys --keyserver hkps://keys.mailvelope.com 9D4B2B6EB8F97156D19669A9FF0812D491B96798 && \
-		gpg --export 9D4B2B6EB8F97156D19669A9FF0812D491B96798 > /usr/share/keyrings/parity.gpg && \
-		echo 'deb [signed-by=/usr/share/keyrings/parity.gpg] https://releases.parity.io/deb release main' > /etc/apt/sources.list.d/parity.list && \
-		apt-get update && \
-		apt-get install -y --no-install-recommends polkadot=${POLKADOT_VERSION#?} && \
+	DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+		libssl1.1 \
+		ca-certificates \
+		curl \
```
Comment (chevdor) [inline scripts/docker/release.Dockerfile:25]: Adding `curl` involves potential security risks. Since `curl` is not required for running `polkadot`, could we either:
- remove the curl dependency entirely ?
- remove `curl` once we are done with it ?

File: scripts/docker/release.Dockerfile [hunk 3]
```diff
@@ -19,24 +19,25 @@ ENV RUST_BACKTRACE 1
...
+		gnupg && \
+	useradd -m -u 1000 -U -s /bin/sh -d /polkadot polkadot && \
+# add repo's gpg keys and install the published polkadot binary
+	gpg --recv-keys --keyserver hkps://keys.mailvelope.com 9D4B2B6EB8F97156D19669A9FF0812D491B96798 && \
```
Comment (chevdor) [inline scripts/docker/release.Dockerfile:29]: Storing `9D4B2B6EB8F97156D19669A9FF0812D491B96798 ` in an `ARG` would bring benefits:
- less repetition
- more readable
- easier/saver to update when needed

**Review by chevdor — CHANGES_REQUESTED** [review]
Dealing with `curl` in all the images would be a good thing security-wise.

File: scripts/docker/release.Dockerfile (hunk 2 above)
Comment (TriplEight) [reply scripts/docker/release.Dockerfile:25]: I do not believe we depend on `curl`, it's there for debug purposes.

Comment (TriplEight) [reply scripts/docker/release.Dockerfile:25]: happy to remove it

File: bridges/deployments/networks/OpenEthereum.Dockerfile (hunk 1 above)
Comment (tomusdrw) [reply bridges/deployments/networks/OpenEthereum.Dockerfile:1]: No I don't think so. If possible please avoid touching `bridges` code in this repo, it's only a clone of the `bridges` repository and the changes are better done there first and then synced here to avoid conflicts.

Is it caught by some blanket CI checks? Can we ignore `bridges` sub-dir from any checks of this kind? In the future the plan is to integrate the bridges code more closely with Polkadot/Substrate repositories, but for now we would rather keep focus on making sure the code works correctly in our repo.

File: scripts/docker/release.Dockerfile (hunk 2 above)
Comment (TriplEight) [reply scripts/docker/release.Dockerfile:25]: let's merge it as is here, I'll create a separate issue for this.

File: scripts/docker/release.Dockerfile (hunk 3 above)
Comment (TriplEight) [reply scripts/docker/release.Dockerfile:29]: let's merge it as is here, I'll create a separate issue for this.

File: bridges/deployments/networks/OpenEthereum.Dockerfile (hunk 1 above)
Comment (TriplEight) [reply bridges/deployments/networks/OpenEthereum.Dockerfile:1]: yeah, a more general question was "what the entire bridges repo does here?" and now I get it, thanks. I'll make a change in bridges then.

Comment (tomusdrw) [reply bridges/deployments/networks/OpenEthereum.Dockerfile:1]: It's a `git subtree`, because we want to use some of the crates from bridges repo, but keep working on the separately. Other option is referencing crates using git repo, but that makes it cumbersome in case of compilation errors (i.e. companion PR system).

File: scripts/docker/release.Dockerfile (hunk 2 above)
Comment (chevdor) [reply scripts/docker/release.Dockerfile:25]: See https://github.com/paritytech/ci_cd/issues/197

File: scripts/docker/release.Dockerfile (hunk 3 above)
Comment (chevdor) [reply scripts/docker/release.Dockerfile:29]: See https://github.com/paritytech/ci_cd/issues/197

**Review by chevdor — APPROVED** [review]

**Review by tomusdrw — APPROVED** [review]
